### PR TITLE
net-misc/netkit-rsh: security fix, add myself as p-m

### PR DESCRIPTION
**Pull request by hlein**
This updates one of our existing patches with Debian's fix for
CVE-2019-7282 and CVE-2019-7283.

Signed-off-by: Hank Leininger <hlein@korelogic.com>
Bug: https://bugs.gentoo.org/717794
Package-Manager: Portage-3.0.20, Repoman-3.0.3

### CODE REVIEW

File: net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+SRC_URI="ftp://ftp.uk.linux.org/pub/linux/Networking/netkit/${P}.tar.gz
+	mirror://gentoo/rexec-1.5.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="pam"
+
+RDEPEND="
+	sys-libs/ncurses:0
```
Comment (thesamesam) [inline net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild:19]: := if it links against it

File: net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+DEPEND="${RDEPEND}"
+BDEPEND="app-arch/xz-utils"
+
+FILECAPS=( cap_net_bind_service usr/bin/r{cp,login,sh} )
+
+src_prepare() {
+	rm -r rexec || die
+	mv ../rexec rexec || die
+
+	eapply "${FILESDIR}"/patches
```
Comment (thesamesam) [inline net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild:30]: Could you convert this to use PATCHES=() in global scope and drop this + eapply_user? (This is just a nit, really).

File: net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		sed -i -e "s|./__conftest|: ./__conftest|" configure || die
+	fi
+}
+
+src_configure() {
+	tc-export CC
+	${CONFIG_SHELL:-/bin/sh} ./configure $(usex pam '' '--without-pam') || die
+
+	sed -i \
+		-e "s:-pipe -O2:${CFLAGS}:" \
```
Comment (thesamesam) [inline net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild:44]: Let's use another [delimiter](https://bugs.gentoo.org/685160) to close [this](https://bugs.gentoo.org/710960) bug.

File: net-misc/netkit-rsh/files/patches/000_all_sectty.patch [hunk 4]
```diff
@@ -0,0 +1,32 @@
+--- netkit-rsh-0.17-pre20000412/rexecd/rexecd.c.sectty	Sat Oct  2 17:50:52 1999
```
Comment (thesamesam) [inline net-misc/netkit-rsh/files/patches/000_all_sectty.patch:1]: This is just to be nice IF you can: can you drop the dates and the directory names at the beginning? Change to a/ and b/?


See here: https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html#clean-patch-howto. Alright if not though, it's a nit.

Comment (thesamesam) [inline net-misc/netkit-rsh/files/patches/000_all_sectty.patch:1]: There are a lot of patches. They already come to 39K on my machine (with the in-tree copy).

The limiter for individual patches is 20K but FILESDIR often grows beyond that: https://qa-reports.gentoo.org/output/filesdir-usage.txt.

Could you pop these in a tarball on e.g. your own git repo or similar? I can do this if you prefer (I suppose it makes review easier).

File: net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild (hunk 2 above)
Comment (thesamesam) [reply net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild:30]: You can do: PATCHES=(
	"${FILESDIR}"/patches/
}

**Review by thesamesam — CHANGES_REQUESTED** [review]
Thank you for working on this!

File: net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild (hunk 1 above)
Comment (hlein) [reply net-misc/netkit-rsh/netkit-rsh-0.17-r12.ebuild:19]: Good call! I'll address this and your other feedback shortly. Sheesh, you volunteer to maintain something, and suddenly are expected to do WORK? =)

File: net-misc/netkit-rsh/files/patches/000_all_sectty.patch (hunk 4 above)
Comment (hlein) [reply net-misc/netkit-rsh/files/patches/000_all_sectty.patch:1]: Well, there's a catch: after this security fix gets accepted and the project gets off the last-rites list, I would really like to update our patchset to/against current Debian, which has other feature-improvements like adding a -p port flag. It's going to be *way* easier to do that and to make clear what has changed, if I'm dropping/modifying individual patches, vs a new opaque tarball that has no clear history from the last one.

If you don't think that'd be a problem, then sure, I can make this a new PATCHVER instead, and ++ again when doing the next set of changes.

Comment (thesamesam) [reply net-misc/netkit-rsh/files/patches/000_all_sectty.patch:1]: Got it. Let's stick with the status quo and remember to do that later.

Comment (thesamesam) [reply net-misc/netkit-rsh/files/patches/000_all_sectty.patch:1]: Oh, and you can add a commit to drop this from package.mask by the way!

**Comment by hlein** [comment]
@thesamesam I think this is ready for another look, thanks!

**Pull request closed without merging**